cfg/builder/builder_finalize.cc

```diff
@@ -453,22 +453,80 @@ vector<UIntSet> CFGBuilder::fillInBlockArguments(core::Context ctx, const CFG::R
     return upperBounds1;
 }
 
-int CFGBuilder::topoSortFwd(vector<BasicBlock *> &target, int nextFree, BasicBlock *currentBB) {
-    // ENFORCE(!marked[currentBB]) // graph is cyclic!
-    if (currentBB->fwdId != -1) {
-        return nextFree;
-    } else {
-        currentBB->fwdId = -2;
-        if (currentBB->bexit.thenb->outerLoops >= currentBB->bexit.elseb->outerLoops) {
-            nextFree = topoSortFwd(target, nextFree, currentBB->bexit.elseb);
-            nextFree = topoSortFwd(target, nextFree, currentBB->bexit.thenb);
```

Comment:
This was added a while ago to ensure that the `then` block would be processed before the `else` block during inference, which ensures that we'd get errors from `then` blocks reported before ones from `else` blocks. I kept this behavior in the new version, because I couldn't find any reason that it would be important to end up processing the `else` block first when iterating over the RPO of the CFG. (Pushing the `then` block onto the work queue before the `else` block means that the `then` block will be added to the post-order after the `else` block, and then will be traversed first when we iterate in reverse during inference.)

I'd be happy to add the conditional back in, but I'm not sure that it's buying us anything other than error reporting order ultimately.

Reply:
We can leave the conditional out; I just wanted to make sure the change to get rid of it was intentional/we were aware of what it did before.